### PR TITLE
Standardize docstrings for table-like inputs

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

The virtualfile_from_data function wrapped in #961 allows for a standardized way for inputting data to PyGMT modules. This change is for standardizing some of the docstrings, specifically for modules that accept 'table-like' vector inputs. The inputs can be of several different PyData object types.

- [numpy.ndarray](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray)
- [pandas.DataFrame](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html#pandas.DataFrame)
- [xarray.Dataset](https://xarray.pydata.org/en/stable/generated/xarray.Dataset.html#xarray.Dataset)
- [geopandas.GeoDataFrame](https://geopandas.org/docs/reference/api/geopandas.GeoDataFrame.html#geopandas.GeoDataFrame) (TODO in #1000)

Plus there will likely be more data type objects wrapped in the future, so it will be good to have a standardized place for them.

TODO:

- [x] info
- [x] plot/plot3d
- [x] rose
- [ ] contour (wait for refactor to use virtualfile_from_data)
- [ ] surface (wait for refactor to use virtualfile_from_data)
- [x] grdtrack
- [ ] blockmean/blockmedian (need to wait for https://github.com/GenericMappingTools/pygmt/issues/1099#issuecomment-804421495)
- [x] velo
- [x] wiggle
- [x] histogram

Addresses https://github.com/GenericMappingTools/pygmt/pull/1000#discussion_r587969541

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/histogram.py
Comment (meghanrjones) [inline pygmt/src/histogram.py:38]: This description should include the list option.

**Review by meghanrjones — COMMENTED** [review]
Would it be worth standardizing the description that accompanies {table-like} in each parameter? e.i., "a 1D/2D  :class:`numpy.ndarray`, a :class:`pandas.DataFrame`, or an :class:`xarray.Dataset` made up of 1D :class:`xarray.DataArray` data variables containing the tabular data."

**Comment by meghanrjones** [comment]
> > Would it be worth standardizing the description that accompanies {table-like} in each parameter? e.i., "a 1D/2D  :class:`numpy.ndarray`, a :class:`pandas.DataFrame`, or an :class:`xarray.Dataset` made up of 1D :class:`xarray.DataArray` data variables containing the tabular data."
> 
> Yes I did think of adding a standard `{table-description}`. The problem is that someone running `help()` on the function in Python would only see the curly brackets and not the expanded description (see image below as an example). The 'Full option list at GMT website' line doesn't get expanded into the full URL in the help docs, it is available only on the online PyGMT docs page.
> 

If there were a `filler_text["table-description"]` similar to `filler_text["table-list"]`, I think would still be printed by `help()` in contrast to the sphinx-dependent documentation (e.g., `:gmt-docs` and `**bold**`). But, it would not be as simple as `filler_text["aliases"]` or `filler_text["table-list"]` to format the docstring with a standard `{table-description}`. So, I am OK with moving forward without that standard replacement text.

Would you prefer to wait on the last three modules or get a final review on this now to merge? It will be at least a few days before I get to the blockm* refactoring.

**Comment by weiji14** [comment]
> If there were a `filler_text["table-description"]` similar to `filler_text["table-list"]`, I think it would still be printed by `help()` in contrast to the sphinx-dependent documentation (e.g., `:gmt-docs` and `**bold**`). But, it would not be as simple as `filler_text["aliases"]` or `filler_text["table-list"]` to format the docstring with a standard `{table-description}`. So, I am OK with moving forward without that standard replacement text.

Hmm, you're right. I'll update things to use a standard {table-description}, will take me an hour or so after lunch.

> Would you prefer to wait on the last three methods or get a final review on this now to merge? It will be at least a few days before I get to the blockm* refactoring.

I think we can do the last three methods next time. Prefer to get this PR in so that I can finalize the geopandas integration PR at #1000. But I'll do the {table-description} issue above :point_up: first.

File: pygmt/helpers/decorators.py
Comment (meghanrjones) [inline pygmt/helpers/decorators.py:238]: It is necessary to remove this period or the period after each {table-classes}.
```suggestion
        "    data variables containing the tabular data"
```

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:1378]: `{table-like}` in this docstring is not rendered.

https://pygmt-git-standardize-table-like-docs-gmt.vercel.app/api/generated/pygmt.clib.Session.virtualfile_from_data.html

Comment (weiji14) [reply pygmt/clib/session.py:1378]: Good catch. Maybe `{table-like}` needs to be placed last? I'll try with this:

```suggestion
        data : None or str or xarray.DataArray or {table-like}
```

Comment (seisman) [reply pygmt/clib/session.py]: The replacement takes place in the fmt_docstring decorator. So I believe you have to add the `@fmt_docstring` decorator before the `virtualfile_from_data` function to make it work.

Comment (weiji14) [reply pygmt/clib/session.py:1379]: You're right, forgot about that! I've added the `@fmt_docstring` decorator in 983dc16676d27f9fc04029191da0e5e01b98f54b so it should render properly now. Had to take away the docstring's curly brackets from the dictionary and f-strings though to prevent it from being formatted :joy:

File: pygmt/helpers/decorators.py
Comment (seisman) [inline pygmt/helpers/decorators.py:234]: I'm OK with the `table-like`, but wondering if we could rewrite it to:
```
filler_text["table-like"] = " or ".join(["numpy.ndarray", "pandas.DataFrame", "xarray.Dataset"])
```

Comment (weiji14) [reply pygmt/helpers/decorators.py]: Good idea! I've done the change at e9a798b6c932cf6134b37c46f4e397ec1e7211a6 and incorporated the other suggested changes too.

**Review by seisman — APPROVED** [review]

**Pull request merged**
